pkg/util/secret.go

```diff
@@ -62,3 +66,35 @@ func UpdateSecret(ctx context.Context, client ctrlruntimeclient.Client, secret *
 	}
 	return false, client.Update(ctx, existing)
 }
+
+type SecretOperationCase string
+
+const (
+	SecretCreate   SecretOperationCase = "create"
+	SecretUpdate   SecretOperationCase = "update"
+	SecretRecreate SecretOperationCase = "recreate"
+)
+
+func OperateOnSecret(secretClient coreclientset.SecretInterface, secret *coreapi.Secret, operationCase SecretOperationCase) error {
```

Comment:
What is the point of this misdirection?

Reply:
Create/Update/ Delete-Create checks are complicated so I thought it is better to have the logic here.

Reply:
updated